packages-ts/starknet/src/utils.ts

```diff
@@ -0,0 +1,207 @@
+  const gateway_url = process.env.NODE_URL || DEVNET_URL
+
+  return { network, gateway_url }
+}
+
+const Funder = (opts: AccountFunderOptions): AccountFunder => {
+  return new AccountFunder(opts)
+}
+
+export const account: FunderOpts = {
+  makeFunderOptsFromEnv: makeFunderOptsFromEnv,
+  Funder: Funder,
+}
+
+interface FundAccounts {
+  account: string
+  amount: number
+}
+
+interface AccountFunderOptions {
+  network?: string
+  gateway_url?: string
+}
+
+// Define the Strategy to use depending on the network.
+class AccountFunder {
+  private opts: AccountFunderOptions
+  private strategy: IFundingStrategy
+
+  constructor(opts: AccountFunderOptions) {
+    this.opts = opts
+    if (this.opts.network === DEVNET_NAME) {
+      this.strategy = new DevnetFundingStrategy()
+      return
+    }
+    this.strategy = new AllowanceFundingStrategy()
+  }
+
+  // This function adds some funds to pre-deployed account that we are using in our test.
+  public async fund(accounts: FundAccounts[]) {
+    await this.strategy.fund(accounts, this.opts)
+  }
+}
+
+interface IFundingStrategy {
+  fund(accounts: FundAccounts[], opts: AccountFunderOptions): Promise<void>
+}
+
+// Fund the Account on Devnet
+class DevnetFundingStrategy implements IFundingStrategy {
+  public async fund(accounts: FundAccounts[], opts: AccountFunderOptions) {
+    accounts.forEach(async (account) => {
+      const body = {
+        address: account.account,
+        amount: account.amount,
+        lite: true,
+      }
+
+      try {
+        await fetch(`${opts.gateway_url}/mint`, {
+          method: 'post',
+          body: JSON.stringify(body),
+          headers: { 'Content-Type': 'application/json' },
+        })
+      } catch (error: any) {}
+    })
+  }
+}
+
+// Fund the Account on Testnet
+class AllowanceFundingStrategy implements IFundingStrategy {
+  public async fund(accounts: FundAccounts[], opts: AccountFunderOptions) {
+    const provider = new SequencerProvider({
+      baseUrl: 'https://alpha4.starknet.io',
+    })
+
+    const keyPair = ec.getKeyPair(process.env.ACCOUNT_PRIVATE_KEY)
+    const accountFunder = new Account(provider, process.env.ACCOUNT.toLowerCase(), keyPair)
+
+    for (const account of accounts) {
+      const estimatFee = await accountFunder.estimateFee({
+        contractAddress: ERC20_ADDRESS_TESTNET,
+        entrypoint: 'transfer',
+        calldata: [
+          account.account,
+          bnToUint256(account.amount).low.toString(),
+          bnToUint256(account.amount).high.toString(),
+        ],
+      })
+      const result = await accountFunder.getNonce()
+      const nonce = toBN(result).toNumber()
+      const hash = await accountFunder.execute(
+        {
+          contractAddress: ERC20_ADDRESS_TESTNET,
+          entrypoint: 'transfer',
+          calldata: [
+            account.account,
+            bnToUint256(account.amount).low.toString(),
+            bnToUint256(account.amount).high.toString(),
+          ],
+        },
```

Comment:
Extract this payload as a variable, instead of c/p here as is also defined couple of lines above.